src/property.c

```diff
@@ -234,6 +236,104 @@ void setter_background_image(Context* context, const char* key, const char* valu
   config_set_str(context->config, key, value);
 }
 
+void setter_uri_schemes(Context* context, const char* key, const char* value)
+{
+  int errorcode;
+  PCRE2_SIZE erroroffset;
+  pcre2_code* code = pcre2_compile(
+    SCHEME_LIST,
+    PCRE2_ZERO_TERMINATED,
+    PCRE2_ANCHORED | PCRE2_CASELESS | PCRE2_ENDANCHORED,
+    &errorcode,
+    &erroroffset,
+    NULL
+  );
+  if (!code) {
+    g_warning("pcre2_compile failed for errorcode `%d` at offset `%d`\n", errorcode, (int)erroroffset);
+    return;
+  }
+
+  // repetitivelly get all schemes in the list, one by one.
+  GSList* schemes = NULL;
+  int scheme_length_sum = 0;
+  while (value[0]) {
+    pcre2_match_data* match_data = pcre2_match_data_create_from_pattern(code, NULL);
+    int res = pcre2_match(
+        code,
+        value,
+        PCRE2_ZERO_TERMINATED,
+        0,
+        PCRE2_ANCHORED | PCRE2_ENDANCHORED | PCRE2_NOTEMPTY,
+        match_data,
+        NULL
+    );
+
+    if (res <= 0) {
+        switch (res) {
+        case 0:
+            printf("Ovector was not big enough. This should not happen.");
+            break;
+        case PCRE2_ERROR_NOMATCH:
+            printf("No match\n");
+            break;
+        default:
+            printf("PCRE2 match error %d\n", res);
+            break;
+        }
+        pcre2_match_data_free(match_data);
+        pcre2_code_free(code);
+        return;
+    }
+
+    PCRE2_SIZE* ovector = pcre2_get_ovector_pointer(match_data);
+    int length = ovector[3] - ovector[2];
+    if (length > 0) {
+        schemes = g_slist_prepend(schemes, g_strndup(value + ovector[2], length)); // get first scheme
+        scheme_length_sum += length + 1; // 1 for separater `|` or terminal null char
+    }
+
+    pcre2_match_data_free(match_data);
+    value = &value[ovector[3] + 1]; // move pointer forward by one word
+  }
+  pcre2_code_free(code);
+
+  // if no schemes specified, remove current regex and return immediately
+  if (scheme_length_sum == 0) {
+    if (context->layout.uri_tag >= 0) {
+      vte_terminal_match_remove(context->layout.vte, context->layout.uri_tag);
+    }
+    return;
+  }
+
+  gchar scheme_pattern[scheme_length_sum];
+  gchar* p = scheme_pattern;
+  for (GSList* scheme = schemes; scheme; scheme = scheme->next) {
+    p = g_stpcpy(p, scheme->data);
+    *p = '|';
+    ++p;
+  }
+  scheme_pattern[scheme_length_sum - 1] = '\0'; // replace last `|` with null char
+  gchar* uri_pattern = g_strconcat("(?:", scheme_pattern, ")", SCHEMELESS_URI, NULL);
+  g_slist_free(schemes);
```

Comment:
The elements of `schemes` is from `g_strndup()` so `g_slist_free_full(schemes, g_free)` would be better.